cmake/modules/xsmm.cmake

```diff
@@ -32,12 +32,12 @@ endif()
 
 set(XSMM_INCLUDE_DIRS ${LIBXSMMROOT}/include)
 
-add_library(xsmm STATIC ${XSMM_SRCS})
-target_include_directories(xsmm PUBLIC ${XSMM_INCLUDE_DIRS})
-target_compile_definitions(xsmm PRIVATE
-  __BLAS=0
+add_mlir_library(xsmm STATIC ${XSMM_SRCS})
+target_include_directories(xsmm PUBLIC
+  $<BUILD_INTERFACE:${XSMM_INCLUDE_DIRS}>
+  $<INSTALL_INTERFACE:include/xsmm>
 )
-add_definitions(-DLIBXSMM_DEFAULT_CONFIG -U_DEBUG)
+add_definitions(-DLIBXSMM_DEFAULT_CONFIG -U_DEBUG -D__BLAS=0)
```

Comment:
@alheinecke Please check this change.

Reply:
My reading is that this disables the BLAS interface in libxsmm. We don't want use this internally, so I guess it's safe to keep it like that for now. In the future, if we have conflicts with other users, we'll need to add an options to enable|disable it from the CMake side.

For now, let's keep it like that. Closing the comment. Thanks!